addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -839,14 +869,134 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
                 } else if (function.equals("volume")) {
                     final int volume = (int) Double.parseDouble(value);
                     updatePlayer(new PlayerUpdateEvent() {
+
                         @Override
                         public void updateListener(SqueezeBoxPlayerEventListener listener) {
                             listener.absoluteVolumeChangeEvent(mac, volume);
                         }
+
                     });
                 }
             }
         }
+
+        private void handleFavorites(String message) {
+            logger.trace("Handle favorites message: {}", message);
+
+            String[] messageParts = message.split("\\s");
+            if (messageParts.length == 2 && "changed".equals(messageParts[1])) {
+                // LMS informing us that favorites have changed; request an update to the favorites list
+                requestFavorites();
+                return;
+            }
+            if (messageParts.length < 7) {
+                logger.trace("No favorites in message.");
+                return;
+            }
+
+            List<Favorite> favorites = new ArrayList<>();
+            String title = "";
+            String count = "";
+            Favorite f = null;
+            for (String part : messageParts) {
+                String id;
+                String name;
+                String type;
+                Boolean isaudio;
+                Boolean hasitems;
+
+                // Title of Favorites
+                if (part.startsWith("title%3A")) {
+                    title = decode(part.substring("title%3A".length()));
+                    logger.trace("Favorite title: {}", title);
+                }
+                // Number of favorites in this message
+                else if (part.startsWith("count%3A")) {
+                    count = part.substring("count%3A".length());
+                    logger.trace("Favorite count: {}", count);
+                }
+                // Favorite ID (in form xxxxxxxxx.n)
+                else if (part.startsWith("id%3A")) {
+                    id = part.substring("id%3A".length());
+                    f = new Favorite(id);
+                    favorites.add(f);
+                }
+                // Favorite name
+                else if (part.startsWith("name%3A")) {
+                    name = decode(part.substring("name%3A".length()));
+                    if (f != null) {
+                        f.name = name;
+                    }
+                }
+                // Favorite type
+                else if (part.startsWith("type%3A")) {
+                    type = decode(part.substring("type%3A".length()));
+                    if (f != null) {
```

Comment:
Is the order guaranteed, will we revisit this if `f` is `null`?

Reply:
Guaranteed is a little strong. :-) 

The order is defined in the LMS code, and is highly unlikely to change at this point. The convention in the LMS code generally is that the id is the first value returned in a response. I was simply trying to be defensive in case of a corrupted response.

The response from the LMS looks like this.
```
favorites = title count favorite1 favorite2 ...

where favoriteN = id name type isaudio hsitems
```